libgnucash/app-utils/gnc-gsettings.cpp

```diff
@@ -823,8 +823,9 @@ void gnc_gsettings_version_upgrade (void)
 
     if (!ogG_maj_min && !og_maj_min) // new install
     {
-        LEAVE("");
-        return;
+        PINFO ("Setting Previous compatibility level to major version minus one: %i", PROJECT_VERSION_MAJOR * 1000 - 1);
+        gnc_gsettings_set_int (GNC_PREFS_GROUP_GENERAL, GNC_PREF_VERSION, PROJECT_VERSION_MAJOR * 1000 - 1);
+        ogG_maj_min = gnc_gsettings_get_user_value (GNC_PREFS_GROUP_GENERAL, GNC_PREF_VERSION);
```

Comment:
On a new install there are no settings to migrate or obsolete. So the only thing to do is setting the current prefs-version. There's no need to go through all other migration steps.

Reply:
I was thinking that if a change was was made for release 5.2, admittedly it may only be a depreciate which does not do much, and I did a new install of 5.4, I would want the previous changes to be processed.

Reply:
Unless it's me I think you may be missing the point. On a *new* install there are *no* settings in GSettings. So there is no history to transform to the current version. If it was an uninstall old version and install new version, that's not a new install in this case. In that case the prefs-version would still be set to 5.2 and hence anything that needs changing to get to 5.4 will be done.

For this to work, an empty prefs-version should be set on first-run to the current version. That would ensure that *future* settings transformations work properly.

Reply:
No It's me, over thinking it, forgot that there would be no preferences on a new install. When you start thinking one way it is sometimes hard to change track. Will set it to the current version.